dv/uvm/core_ibex/sim.py

```diff
@@ -190,6 +193,74 @@ def get_simulator_cmd(simulator, yaml_path, enables):
             subst_cmd(entry['sim']['cmd'], enables, entry['sim'], env_vars))
 
 
+def filter_tests_by_config(cfg, test_list):
+    '''Filter the provided testlist to ensure all required RTL parameters are
+       provided by the Ibex core configuration.
+
+    Arguments:
+
+        cfg: string name of the ibex config being tested, should match a
+             config name from ibex_configs.yaml.
+
+        test_list: list of test entry objects parsed from the YAML testlist
+
+    Returns:
+
+        filtered_test_list: a list of test entry objects, filtered such that
+                            all tests incompatible with the specified ibex
+                            config have been removed.
+
+                            e.g. if the "small" config has been specified, this
+                            function will filter out all tests that require
+                            B-extension and PMP parameters
+    '''
+    filtered_test_list = []
+    config = parse_config(cfg, os.path.join(_IBEX_ROOT, "ibex_configs.yaml"))
+
+    for test in test_list:
+        if "rtl_params" in test:
+            param_dict = test['rtl_params']
+            matching_params = False
```

Comment:
I think I'd consider moving the stuff that sets `matching_params` into a separate function. It'll get rid of some of the indentation and I think the code will become a bit clearer (you're essentially just filtering a list, and that will look much more obvious then, I think).